### ISSUE TITLE
Unauthorized Error with Patreon When Using Non-Creating Account

### ISSUE THREAD

**Issue by Mike-E-angelo** [aspnet-contrib/AspNet.Security.OAuth.Providers#725]
(This may be a Patreon-related problem so have reported there as well -- but I wanted to get the ball rolling here in case there is something that can be done here)

### Describe the bug

I am running into an issue when using Patreon.  If I sign into my Oauth2 application with the Patreon account that created the Patreon Oauth2 application, everything works OK. 

However, when signing in with a Patreon account that did not create the Oauth2 account, I get unauthorized errors thrown on `/signin-patreon`

Here's the message:
```
"The server could not verify that you are authorized to access the URL requested. You either supplied the wrong credentials (e.g. a bad password), or your browser doesn't understand how to supply the credentials required."
```

### Steps To reproduce

You may attempt to login via Patreon here and witness the glory of the 500 error yourself:
https://alpha.starbeam.one/Identity/Account/Login?ReturnUrl=%2F

### Expected behaviour

No errors :)

### Actual behaviour

500 Error is encountered for accounts that did not create the Patreon Oauth2 registration/application.  The account that created the Patreon Oauth2 registration/application does not run into this issue.

### System information

- OS: Windows 10
- Library Version : 6.0.11
- .NET version (e.g. output from `dotnet --info`) - net6.0 hosted in Azure AppService

### Additional context

Thank you for any assistance 🙏

**Comment by martincostello** [comment]
An initial guess at what this might be based purely on the description is that maybe it's a scopes thing?

**Comment by Mike-E-angelo** [comment]
That's a good point @martincostello I am requesting the `campaigns` scope but what if the account isn't a creator and doesn't have campaigns? 🤔

**Comment by kevinchalet** [comment]
Did you hear back from Patreon, @Mike-E-angelo?

**Comment by Mike-E-angelo** [comment]
It took them an entire week to reply and tell me that I was in the wrong place 😭  Opened a thread here:
https://www.patreondevelopers.com/t/oauth2-troubles/6324

I'll open this thread if I require anything else here 👍

**Comment by Mike-E-angelo** [comment]
FWIW as you might expect this was due to user error. 🤦‍♂️

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Patreon/PatreonAuthenticationOptions.cs#L26

I was clearing scopes, and then setting to `"campaigns"` not realizing that `identity` was supposed to be set as well.  

Interesting to note that it appears `identity` is set to the creator of the Oauth2 application regardless.  That is, everything works as expected when I log in with the account that created the Patreon Oauth2 application.  This is how I initially tested my environment and it led me to believe everything is configured correctly.  

It is only when I attempted to login with a different account did I experience the issue here.
